rich/_timer.py

```diff
@@ -6,10 +6,11 @@
 from time import time
 
 import contextlib
+from typing import Iterator
 
 
 @contextlib.contextmanager
-def timer(subject: str = "time"):
+def timer(subject: str = "time") -> Iterator[None]:
```

Comment:
I think `typing.Generator` would be more accurate here.

Reply:
https://github.com/python/typeshed/blob/master/stdlib/contextlib.pyi#L25 shows it takes in an iterator.